Feat google authentication

DongNguyenQ opened the pull request:
# Feature 
- Google signin

Commit:
Add map key, setup bloc, signup screen

Commit:
On going

Commit:
On going

Commit:
- Fix No firebase app configure [00005]
- Change applicationId, bundleId to .dev

Commit:
On fixing widget test

Commit:
- Clean up code
- Add Localisation

Commit:
Pull from base repository

Commit:
- Pull update from base repository
- Revert config to default

suesitran commented on ios/Runner.xcodeproj/project.pbxproj:
```diff
@@ -47,7 +47,7 @@
 		97C147001CF9000F007C117D /* Base */ = {isa = PBXFileReference; lastKnownFileType = file.storyboard; name = Base; path = Base.lproj/LaunchScreen.storyboard; sourceTree = "<group>"; };
 		97C147021CF9000F007C117D /* Info.plist */ = {isa = PBXFileReference; lastKnownFileType = text.plist.xml; path = Info.plist; sourceTree = "<group>"; };
-		E2CFCC9254DDE1C53A1BBADF /* GoogleService-Info.plist */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.plist.xml; name = "GoogleService-Info.plist"; path = "Runner/GoogleService-Info.plist"; sourceTree = "<group>"; };
+		9B4865F85BCB56F3A8EF563F /* GoogleService-Info.plist */ = {isa = PBXFileReference; includeInIndex = 1; lastKnownFileType = text.plist.xml; name = "GoogleService-Info.plist"; path = "Runner/GoogleService-Info.plist"; sourceTree = "<group>"; };
```
please do not change xcode settings. Please revert this file to its state in dev

DongNguyenQ committed:
- Add test nartus ui
- Update gitignore nartus ui package

suesitran commented on lcov.base.info:
```diff
@@ -0,0 +1,202 @@
+DA:25,0
+DA:26,0
+DA:28,0
+LF:7
```
Please do not include this file in the commit. Maybe I’ll need to update gitignore to ignore this file

suesitran commented on lib/constants/dimens.dart:
```diff
@@ -4,4 +4,6 @@ class Dimension {
 
   static const double spacing16 = 16.0;
+
+  static const double screenEdgeSpacing = 12.0;
```
Just use the name spacing12 here

suesitran commented on lib/features/signup/signup_screen.dart:
```diff
@@ -0,0 +1,129 @@
+          ),
+          Gap.v12(),
+          SizedBox(
+            width: double.infinity,
```
If width  is infinity, u don’t need SizedBox here

suesitran commented on lib/features/signup/signup_screen.dart:
```diff
@@ -0,0 +1,129 @@
+      context.read<GoogleSignupBloc>().add(SignUpByGoogleEvent());
+}
+
+Future<T?> _navToHomeScreen<T>(BuildContext context) {
```
Just delete unimplemented methods for now, we do not want app to crash during testing because of these exception

suesitran commented on packages/nartus_authentication/lib/src/authentication_service.dart:
```diff
@@ -0,0 +1,64 @@
+    if (firebaseUser == null) throw AuthenticateFailedException.userNotFound();
+
+    return UserDetail(
+      name: firebaseUser.displayName!,
```
Instead of unsafe null here, it’s better to use nullable in name and photo url

suesitran reviewed (changes requested):
Partially review

suesitran commented on packages/nartus_authentication/test/nartus_authentication_test.dart:
```diff
@@ -0,0 +1,41 @@
+    });
+  });
+
+  test('given user signing in by google, when user cancel login, then throw AuthenticateFailedException.userCancelled', () {
```
duplicated test with 'given user signing in by google, when user cancel login, then throw AuthenticateFailedException.userCancelled'

suesitran commented on packages/nartus_authentication/test/nartus_authentication_test.dart:
```diff
@@ -0,0 +1,41 @@
+      when(googleSignIn.signIn())
+          .thenAnswer((realInvocation) => Future.value(null));
+
+      expect(() => authService.signinGoogle(), throwsA(isA<AuthenticateFailedException>()));
```
what is this test different from above test?

suesitran commented on packages/nartus_authentication/test/nartus_authentication_test.dart:
```diff
@@ -0,0 +1,41 @@
+    when(googleSignIn.signIn())
+        .thenAnswer((realInvocation) => throw PlatformException(code: 'sign_in_canceled'));
+
+    expect(() => authService.signinGoogle(), throwsA(isA<AuthenticateFailedException>()));
```
where is the exception with AuthenticateFailedException.userCancelled?

suesitran commented on packages/nartus_ui_package/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java:
```diff
@@ -0,0 +1,23 @@
+package io.flutter.plugins;
```
please add this file to .gitignore

suesitran commented on packages/nartus_ui_package/ios/Flutter/Generated.xcconfig:
```diff
@@ -0,0 +1,13 @@
+// This is a generated file; do not edit or check into version control.
```
please remove this file

suesitran commented on packages/nartus_ui_package/ios/Flutter/flutter_export_environment.sh:
```diff
@@ -0,0 +1,13 @@
+#!/bin/sh
```
please remove this file

suesitran commented on packages/nartus_ui_package/ios/Runner/GeneratedPluginRegistrant.h:
```diff
@@ -0,0 +1,19 @@
+//
```
please remove this file

suesitran commented on packages/nartus_ui_package/ios/Runner/GeneratedPluginRegistrant.m:
```diff
@@ -0,0 +1,14 @@
+//
```
please remove this file

suesitran commented on packages/nartus_ui_package/lib/widgets/buttons.dart:
```diff
@@ -0,0 +1,23 @@
+      onPressed: onPressed,
+      label: isBusy! ? const CircularProgressIndicator() : Text(text),
+      style: TextButton.styleFrom(
+          maximumSize: const Size(double.infinity, 46),
```
setting max height will limit the scaling when large text is ON, not a good practice to do

suesitran commented on packages/nartus_ui_package/macos/Flutter/ephemeral/Flutter-Generated.xcconfig:
```diff
@@ -0,0 +1,11 @@
+// This is a generated file; do not edit or check into version control.
```
please remove this file

suesitran commented on packages/nartus_ui_package/macos/Flutter/ephemeral/flutter_export_environment.sh:
```diff
@@ -0,0 +1,12 @@
+#!/bin/sh
```
please remove this file

suesitran commented on packages/nartus_ui_package/test/widgets/buttons_test.dart:
```diff
@@ -0,0 +1,56 @@
+          text: 'Test button', onPressed: () {});
+
+        await tester.pumpWidget(MaterialApp(home: textButton,));
+        await tester.pumpAndSettle();
```
you can use `wrapAndPump` here too

suesitran commented on packages/nartus_ui_package/test/widgets/buttons_test.dart:
```diff
@@ -0,0 +1,56 @@
+        );
+
+        await tester.pumpWidget(MaterialApp(home: textButton));
+        await tester.pump();
```
you can use `wrapAndPump` here

suesitran commented on packages/nartus_ui_package/test/widgets/buttons_test.dart:
```diff
@@ -0,0 +1,56 @@
+        );
+
+        await tester.pumpWidget(MaterialApp(home: textButton));
+        await tester.pumpAndSettle();
```
you can use `wrapAndPump` here

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+
+      /// We need to wrap gap with some widget to prevent SizedBox take all available space
+      await tester.pumpWidget(MaterialApp(home: Column(children: [gapV20],),));
+      await tester.pumpAndSettle();
```
you can use `wrapAndPump` here

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+      await tester.pumpWidget(MaterialApp(home: Column(children: [gapV20],),));
+      await tester.pumpAndSettle();
+
+      final Size size = gapKey.currentContext!.size!;
```
you should check the height of SizedBox widget that's being rendered, instead of checking the size of created widget

```
        final SizedBox size = tester.widget<SizedBox>(find.byType(SizedBox));
        expect(size.height, 16);
```

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+
+        final Size size = gapKey.currentContext!.size!;
+
+        expect(size.height, 16);
```
same comment above

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+
+        final Size size = gapKey.currentContext!.size!;
+
+        expect(size.height, 12);
```
same comment above

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+
+        final Size size = gapKey.currentContext!.size!;
+
+        expect(size.height, 08);
```
same comment above

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+
+        final Size size = gapKey.currentContext!.size!;
+
+        expect(size.height, 04);
```
same comment above

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+
+        final Size size = gapKey.currentContext!.size!;
+
+        expect(size.width, 20);
```
same comment above

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+
+          final Size size = gapKey.currentContext!.size!;
+
+          expect(size.width, 16);
```
same comment above

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+
+        final Size size = gapKey.currentContext!.size!;
+
+        expect(size.width, 12);
```
same comment above

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+
+        final Size size = gapKey.currentContext!.size!;
+
+        expect(size.width, 08);
```
same comment above

suesitran commented on packages/nartus_ui_package/test/widgets/gaps_test.dart:
```diff
@@ -0,0 +1,149 @@
+        final Size size = gapKey.currentContext!.size!;
+
+        expect(size.width, 04);
+      });
```
same comment above

suesitran commented on pubspec.lock:
```diff
@@ -265,6 +265,13 @@ packages:
       url: "https://pub.dartlang.org"
```
please remove this file from commits

suesitran commented on test/features/signup/signup_screen_test.dart:
```diff
@@ -0,0 +1,81 @@
+          'Continue with Google'
+        );
+
+        await widgetTester.blocWrapAndPump<GoogleSignupBloc>(mockSignUpBloc, screen, infiniteAnimationWidget: true);
```
`IDSignUp` screen has its own `GoogleSignupBloc`, so `blocWrapAndPump` does not provide mocked bloc to it.

Mocking `GoogleSignupBloc` is not needed here if you're only checking existence of `googleButton`

suesitran commented on test/mock_firebase.dart:
```diff
@@ -0,0 +1,18 @@
+// found in the LICENSE file.
+
+
+/// Source : https://stackoverflow.com/questions/63662031/how-to-mock-the-firebaseapp-in-flutter
```
great to put comment of reference link, but can you remove the copyright? or change it to my name :)

suesitran commented on lib/features/signup/signup_screen.dart:
```diff
@@ -0,0 +1,129 @@
+          SizedBox(
+            width: double.infinity,
+            child: Text(
+              'Welcome to Interactive Dairy',
```
should we move this into localisation?

suesitran commented on lib/features/signup/signup_screen.dart:
```diff
@@ -0,0 +1,129 @@
+          Gap.v12(),
+          SizedBox(
+            width: double.infinity,
+            child: Text("Let's join us to create your journey"),
```
should we move this into localisation?

suesitran commented on lib/features/signup/signup_screen.dart:
```diff
@@ -0,0 +1,129 @@
+    return BlocConsumer<GoogleSignupBloc, GoogleSignupState>(
+      builder: (BuildContext stateContext, GoogleSignupState googleState) {
+        return IDButton(
+          text: 'Continue with Google',
```
should we move this into localisation?

suesitran commented on lib/features/signup/signup_screen.dart:
```diff
@@ -0,0 +1,129 @@
+  @override
+  Widget build(BuildContext context) {
+    return IDButton(
+      text: 'Continue with Apple', onPressed: () => _signUpByApple(context));
```
should we move this into localisation?

suesitran commented on lib/features/signup/signup_screen.dart:
```diff
@@ -0,0 +1,129 @@
+        children: <Widget>[
+          IDTextField(
+            controller: emailCtrl,
+            hint: 'Email',
```
should we move this into localisation?

suesitran commented on lib/features/signup/signup_screen.dart:
```diff
@@ -0,0 +1,129 @@
+          const Gap.v20(),
+          IDTextField(
+            controller: passCtrl,
+            hint: 'Password',
```
should we move this into localisation?

suesitran commented on ios/Runner/AppDelegate.swift:
```diff
@@ -1,13 +1,15 @@
     _ application: UIApplication,
     didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?
   ) -> Bool {
+      FirebaseApp.configure()
```
i dont think this change is needed, `Firebase.initializeApp()` can be done in dart

suesitran committed:
Ignore all lcov

DongNguyenQ committed:
- Change dimension variable

DongNguyenQ committed:
Merge branch 'dev' of https://github.com/DongNguyenQ/interactive_diary into dev

DongNguyenQ committed:
- Allow nullable user name and photo url
- Fix test gaps_test to test rendered Gap
- Remove copy right from mock firebase
- Remove duplicated test user canceled signin of authentication package test
- Remove UnImplementedError from signup screen

DongNguyenQ committed:
Merged newest changes from Suesi repository

DongNguyenQ committed:
Merge branch 'feat-google-authentication' of https://github.com/DongNguyenQ/interactive_diary into feat-google-authentication

DongNguyenQ committed:
Update gitignore

DongNguyenQ committed:
Remove cached to apply new gitignore
